.ci/generate.jenkinsfile

```diff
@@ -0,0 +1,367 @@
+
+// Maybe parameterized somewhere
+dockerUsername = 'center-c3i-docker.jfrog.io'
+dockerhubUsername = 'conanio'
+
+
+def parseVersion(String version) {
+    def matcher = (version =~ /(\d+).(\d+).(\d+)/)
+    return [matcher[0][1] as String, matcher[0][2] as String, matcher[0][3] as String]
+}
+
+String getVersionMajor(String version) {
+    def (major, _1, _2) = parseVersion(version)
+    return major
+}
+
+
+def buildImage(String latestMaster, String latestBranch, String expectedImage, String buildArgs, String target = null, String extraCacheFrom = null) {
+    withCredentials([usernamePassword(credentialsId: 'center-c3i-docker', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {
+        sh 'docker login --username $USERNAME --password $PASSWORD ' + "${dockerUsername}"
+    }
+
+    String cacheFrom = extraCacheFrom ? " --cache-from ${extraCacheFrom}" : ''
+    if (0 == sh(script: "docker pull ${latestMaster}", returnStatus: true)) {
+        cacheFrom = "${cacheFrom} --cache-from ${latestMaster}"
+    }
+    if (0 == sh(script: "docker pull ${latestBranch}", returnStatus: true)) {
+        cacheFrom = "${cacheFrom} --cache-from ${latestBranch}"
+    }
```

Comment:
maybe avoid this cacheFrom at all?

Reply:
We use cache-from because we re-use the same base docker image for last stage. It's not something like `--no-cache` for Docker build, which consume all stuff from local cache, but it means, consume from that specific image.